optuna/_transform.py

```diff
@@ -158,6 +158,9 @@ def untransform(self, trans_params: numpy.ndarray) -> Dict[str, Any]:
 
         return params
 
+    def is_same_search_space(self, search_space: Dict[str, BaseDistribution]) -> bool:
+        return dict(self._search_space) == search_space
```

Comment:
I'm not sure if we should define this logic here. We can also define this as a private function in `optuna/samplers/_cmaes.py` What do you think?